python.pkgs.sane: init at 2.8.2

doronbehar opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

Needed for https://gitlab.gnome.org/GNOME/ocrfeeder which I tend to package afterwards.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

doronbehar committed:
python.pkgs.sane: init at 2.8.2

jonringer commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5608,6 +5608,8 @@ in {
 
   safe = callPackage ../development/python-modules/safe { };
 
+  sane = callPackage ../development/python-modules/sane { };
```
future proof against a package of the same name being added python-modules
```suggestion
  sane = callPackage ../development/python-modules/sane {
    inherit (pkgs) saneBackends;
  };
```

doronbehar replied:
:+1:.

jonringer commented on pkgs/development/python-modules/sane/default.nix:
```diff
@@ -0,0 +1,27 @@
+  meta = with lib; {
+    homepage = "https://github.com/python-pillow/Sane";
+    description = "Python interface to the SANE scanner and frame grabber ";
+    license = "https://github.com/python-pillow/Sane/blob/master/COPYING";
```
this is MIT-like, but some of the wording is slightly different.
but a typical license usually has many attributes:
```
nix-repl> lib.licenses.mit
{ fullName = "MIT License"; shortName = "mit"; spdxId = "MIT"; url = "http://spdx.org/licenses/MIT.html"; }
```

doronbehar replied:
Great, didn't know what to put there.

jonringer commented on pkgs/development/python-modules/sane/default.nix:
```diff
@@ -0,0 +1,27 @@
+  buildInputs = [
+    saneBackends
+  ];
+
```
please add tests:
```
copying build/lib.linux-x86_64-3.7/_sane.cpython-37m-x86_64-linux-gnu.so ->

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

doronbehar replied:
:+1:.

jonringer commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5608,6 +5608,10 @@ in {
 
   safe = callPackage ../development/python-modules/safe { };
 
+  sane = callPackage ../development/python-modules/sane {
```
this should probably match the pypi name, so it feels similar to a typical setup.py or requirements.txt
```suggestion
  python-sane = callPackage ../development/python-modules/python-sane {
```

cc @FRidh thoughts?

doronbehar replied:
I think they named the project `python-sane` to differentiate it from what we call `saneBackends`. They name themselves just "Sane" on GitHub: https://github.com/python-pillow/Sane

doronbehar replied:
Last push is to mark it as python3 only.

SuperSandro2000 commented:
@doronbehar please resolve the merge conflict.

doronbehar commented:
Fixed merge conflicts and rebased with the commit of this PR in https://github.com/NixOS/nixpkgs/pull/84939 - this is a dep for the main package added in https://github.com/NixOS/nixpkgs/pull/84939 .
